### ISSUE TITLE
SC2142 occurs when parameters are used in a function

### ISSUE THREAD

**Issue by HaleTom** [koalaman/shellcheck#1263]
#### For bugs
- SC2142
- My shellcheck version: online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
alias f='f(){ echo You said: "$@"; }; f'
```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
 
Line 2:
alias f='f(){ echo You said: "$@"; }; f'
      ^-- SC2142: Aliases can't use positional parameters. Use a function.

$ 
```

#### Here's what I wanted or expected to see:

I'd expect to see nothing in this specific case. I am already using a function.

Aliases *can* use positional parameters with a function, see [this answer](https://stackoverflow.com/a/42466441/5353461).

I wrote that and would be happy if you borrowed from it to update the explanation of this check.

**Comment by koalaman** [comment]
You're right of course. I've updated the wiki to be more explicit about uses where positional parameters does not  intend to refer to directly refer to the alias's parameters.

**Comment by HaleTom** [comment]
Would you consider disabling the warning if the alias starts with a function definition?

**Comment by HaleTom** [comment]
I have several real-world use cases where this is a false positive, eg:

```
alias pless='f(){ pygmentize -f terminal256 -O style=monokai -g "$@" | less -L; };f'
```

Could you please re-open this?

**Comment by koalaman** [comment]
It could potentially be recognized, but why are you doing it this way instead of `pless(){ pygmentize -f terminal256 -O style=monokai -g "$@" | less -L; }`?
